src/main/java/org/datavyu/plugins/ffmpeg/ImageCanvasPlayerThread.java

```diff
@@ -29,25 +29,76 @@
     private static final double REFRESH_PERIOD = 0.01; // >= 1/fps
     private static final double TO_MILLIS = 1000.0;
 
+    /** x1 and y1 are respectively the x and y coordinates of the first corner of the destination rectangle. */
+    private int x1, y1;
+
+    /** x2 and y2 are respectively the x and y coordinates of the second corner of the destination rectangle. */
+    private int x2, y2;
+
     ImageCanvasPlayerThread(MediaPlayerData mediaPlayerData) {
         this.mediaPlayerData = mediaPlayerData;
         setName("Ffmpeg image canvas player thread");
         setDaemon(false);
     }
 
-    private void updateDisplay() {
-        //long time = System.nanoTime();
+    /**
+     * Draws as much of the specified area of the specified image as is
+     * currently available, scaling it on the fly to fit inside the
+     * specified area of the destination drawable surface
+     * specified by x1, y1, x2, y2 variables.
+     */
+    private synchronized void updateDisplay() {
+        Dimension size;
+        /** This loop is replacing the component listener used
+         * to update the image while resizing the canvas, listener
+         * is causing flickering during resizing
+         */
         do {
+            size = canvas.getSize();
+            if (size == null) { return; }
             do {
-                Graphics graphics = strategy.getDrawGraphics();
-                graphics.drawImage(image, 0, 0, canvas.getWidth(), canvas.getHeight(),  null);
-                graphics.dispose();
-            } while (strategy.contentsRestored());
-            strategy.show();
-        } while (strategy.contentsLost());
-        //System.out.println("Time to display took: " + (System.nanoTime() - time)/1e6 + " ms");
+                do {
+                    Graphics graphics = strategy.getDrawGraphics();
+                    if (graphics == null) { return; }
+                    scaleImage(); // calculate the coordinate of the scaled display rectangle
+                    graphics.drawImage(image, x1, y1, x2, y2, 0, 0, this.width, this.height, null);
+                    graphics.dispose();
+                } while (strategy.contentsRestored());
+                strategy.show();
+            } while (strategy.contentsLost());
+            // Repeat the rendering if the target changed size
+        } while (!size.equals(canvas.getSize()));
+    }
+
+    /**
+     * Calculate the coordinates of the destination scaled rectangle
```

Comment:
Unclear comment

Reply:
Is this comment better?
"Scale an image by calculating the coordinates of the target image to be rendered in the canvas and maintaining the aspect ratio of the source image."